### ISSUE TITLE
Using a bunch of Keras models in a zip file for Dl4j

### ISSUE THREAD

**Issue by sandylunar** [deeplearning4j/deeplearning4j#9293]
#### Feature Request

I've converted a bunch of Keras models into the Dl4j .zip format for WekaDeeplearning4j, a Weka wrapper for Dl4j. I was wondering whether the Dl4j maintainers would be interested in using these zip files for Dl4j? It would expand the Dl4j Model Zoo substantially.
- Keras DenseNet121
- Keras DenseNet169
- Keras DenseNet201
- Keras EfficientNetB0
- Keras EfficientNetB1
- Keras EfficientNetB2
- Keras EfficientNetB3
- Keras EfficientNetB4
- Keras EfficientNetB5
- Keras EfficientNetB6
- Keras EfficientNetB7
- Keras InceptionV3
- Keras NASNetLarge
- Keras NASNetMobile
- Keras ResNet101
- Keras ResNet101V2
- Keras ResNet152
- Keras ResNet152V2
- Keras ResNet50
- Keras ResNet50V2
- Keras VGG16
- Keras VGG19
- Keras Xception

You can download them from [here](https://github.com/Waikato/wekaDeeplearning4j/releases/tag/zoo-models)

I ask because I don't have the capacity to code up the architectures in Dl4j for each one of them, so the model architectures would only be available in pretrained format, but maybe that's ok (up to the maintainers I suppose).

#### Additional Information

This is automatically mined by [FRMiner2.0](https://github.com/elloworl/FRMiner2.0) from Gitter. 
Rhys Compton @basedrhys posted to Room [eclipse/deeplearning4j](https://gitter.im/eclipse/deeplearning4j?at=607e5880d7953918616b819c) on Apr 20 12:28 (UTC+8).

**Comment by sandylunar** [comment]
Our work aims to mine useful information from live chats, and let the team be aware of them. Hopefully this can help somehow.

**Comment by agibsonccc** [comment]
@sandylunar  yes we would definitely be interested! Could you submit a pull request to the deeplearning4j model zoo? Steps for how to integrate would also be appreciated. Our model zoo currently leverages an azure blob storage setup. If all we need to do is upload those then that would work fine. Anything we should need to know should be discussed in the issue. Thanks!

**Comment by sandylunar** [comment]
> @sandylunar yes we would definitely be interested! Could you submit a pull request to the deeplearning4j model zoo? Steps for how to integrate would also be appreciated. Our model zoo currently leverages an azure blob storage setup. If all we need to do is upload those then that would work fine. Anything we should need to know should be discussed in the issue. Thanks!

Not sure how to submit PR to deeplearning4j model zoo, but we will try for that, and see if that's all you need.

**Comment by basedrhys** [comment]
@sandylunar i can have a go at doing the PR if you like - there’s a few gotchas in using these models to be aware of.

**Comment by sandylunar** [comment]
> @sandylunar i can have a go at doing the PR if you like - there’s a few gotchas in using these models to be aware of.

Definitely yes. Our tool find your utterances are quite useful in Gitter. We just move them here. So It will be much more appropriate to do the PR by yourself :)

**Comment by agibsonccc** [comment]
@sandylunar  @basedrhys  JFYI for mdoels from the default deeplearning4j model zoo, we would need to upload them to the normal deepleanring4j azure cdn. I would be happy to accept the models via something like google drive and help participate in making this a smooth contribution.

**Comment by agibsonccc** [comment]
@basedrhys just following up here, we're getting ready for a 1.0 and I was wondering if we could get this contribution in there.  Thanks!

**Comment by basedrhys** [comment]
Hey @agibsonccc sorry for the delay. I unfortunately am having trouble getting DL4J setup for dev locally and don't have the capacity to complete this task at the moment. The best I can do is point you to the [download link](https://github.com/Waikato/wekaDeeplearning4j/releases/tag/zoo-models), and some links showing their usage

* [KerasResNet Class](https://github.com/Waikato/wekaDeeplearning4j/blob/master/src/main/java/weka/dl4j/zoo/KerasResNet.java)
* [KerasEfficientNet Class](https://github.com/Waikato/wekaDeeplearning4j/blob/master/src/main/java/weka/dl4j/zoo/KerasEfficientNet.java) (note it's loaded with channels last (NHWC) instead of the usual NCWH
* [KerasZooModel Base Class](https://github.com/Waikato/wekaDeeplearning4j/blob/fb0a015c4d2ce4929e62f1f28b09862b7d554f17/src/main/java/weka/dl4j/zoo/keras/KerasZooModel.java#L38) (essentially cloning the DL4J ZooModel class), this does the loading from the downloaded .zip file
* [All Download Links and Checksums](https://github.com/Waikato/wekaDeeplearning4j/blob/fb0a015c4d2ce4929e62f1f28b09862b7d554f17/src/main/java/weka/dl4j/zoo/keras/KerasConstants.java#L11)

Let me know how you get on, apologies I can't be of more help.

**Comment by agibsonccc** [comment]
@basedrhys was your problem related to the shaded modules? If that was the case, I would like your feedback on this page: https://deeplearning4j.konduit.ai/multi-project/how-to-guides/import-in-to-your-favorite-ide even a quick skim would help a lot so this doesn't happen again. I don't like that people have pain contributing. That would be a bigger help than solving the github issue here.

On that front, I like the idea of the model zoo architecture your team came up with for the keras models. In the upcoming roadmap we'll be doing a model zoo in a similar capacity for samediff. If you don't mind, I would like to borrow some of the ideas here. Beyond that, I can figure out a smaller abstraction that just transparently downloads/loads the weights for a given framework that then just gives the necessary conversion like what your team does with some of the models. Thanks!

**Comment by agibsonccc** [comment]
Closing — this is out of scope for the Keras import module. Keras import supports HDF5 functional/sequential models; nested subclass models and export-to-Keras are not planned.
